### PR TITLE
samd: Add some documentation for the extended samd port.

### PR DESCRIPTION
**Pull request by robert-hh**
Matching PR #8704.  It consists of the basic introduction, a quickref document and a pinout document, showing, which MCU device is accessible at which board pin.
I hope the work is not futile.

### CODE REVIEW

File: docs/samd/pinout.rst [hunk 1]
```diff
@@ -0,0 +1,681 @@
...
+which is a subset of the MCU's Pin MUX table and is specific to each board, as the
+available pin set varies. The structure of the table is the same for each board, but
+the set of rows is different.
+
+.. _samd21_pinout_table:
+
+Adafruit ItsyBitsy M0 Express pin assignment table
+--------------------------------------------------
+
+|
```
Comment (dpgeorge) [inline docs/samd/pinout.rst:17]: What do these "|" characters do?  They aren't used elsewhere in the docs (except for mimxrt, which I think you also wrote).

**Comment by dpgeorge** [comment]
Please can you make sure the docs CI passes.

**Comment by robert-hh** [comment]
Done. It was the second-to-last commit, missing indents at a code block.

File: docs/samd/pinout.rst (hunk 1 above)
Comment (dpgeorge) [reply docs/samd/pinout.rst:17]: But shouldn't you let Sphinx/HTML/latex decide in the spacing and layout?

**Comment by robert-hh** [comment]
Helle @dpgeorge The new commit removes the vertical bars in the file. Once you merge the samd_improve branch, the pinout file has to be update. I have prepared that already, so it can be swapped in any time.

**Comment by dpgeorge** [comment]
Squashed all commits into one (most were fixes/improvements) and merged in 11910e2fa1c8f5dc76cbc90598ff81526d1f6312 with some minor formatting fixes.

Thank you very much!

**Comment by robert-hh** [comment]
Thanks. Yes, the commits were all around that single topic "documentation". The samd port looks now pretty complete. There is little left. Since most boards do not have a on-board SD adapter, I skipped I2S support. dac timed and adc timed are there as a "poor man's audio interface", so you can play and record short audio snippets at moderate quality. The mcu's have hardware support for position decoder and counters, but since there is no common approach to that yet, it has to wait.
b.t.w.: There is a PR21 open at the micropython-media repository for the missing pictures. But that's not urgent at all.

**Comment by andypiper** [comment]
Amazing work here. Thanks @robert-hh 🙂

**Pull request closed without merging**